tests/yam/agama/boot_agama.pm

```diff
@@ -53,7 +53,11 @@ sub run {
     my $agama_up_an_running = $testapi::distri->get_agama_up_an_running();
 
     # prepare kernel parameters
+    if (my $agama_profile = get_var('AGAMA_PROFILE')) {
+        expand_agama_profile($agama_profile);
+    }
     if (my $agama_auto = get_var('AGAMA_AUTO')) {
+        expand_agama_profile($agama_auto);
```

Comment:
this is weird, because we use `expand_variables` in `expand_profile_url` so now it is time to separate agama from autoyast, I would recommend to rename `expand_profile_url` to `expand_agama_profile` and create `expand_agama_variables` to do those replacements of type, `$profile =~ s/\{\{$var\}\}/$value/g;`

Reply:
updated